src/SAML2/SOAP.php

```diff
@@ -18,17 +19,24 @@ class SOAP extends Binding
      */
     public function send(Message $message)
     {
-        header('Content-Type: text/xml', true);
-        $outputFromIdp = '<?xml version="1.0" encoding="UTF-8"?>';
-        $outputFromIdp .= '<SOAP-ENV:Envelope xmlns:SOAP-ENV="http://schemas.xmlsoap.org/soap/envelope/">';
-        $outputFromIdp .= '<SOAP-ENV:Body>';
-        $xmlMessage = $message->toSignedXML();
-        Utils::getContainer()->debugMessage($xmlMessage, 'out');
-        $tempOutputFromIdp = $xmlMessage->ownerDocument->saveXML($xmlMessage);
-        $outputFromIdp .= $tempOutputFromIdp;
-        $outputFromIdp .= '</SOAP-ENV:Body>';
-        $outputFromIdp .= '</SOAP-ENV:Envelope>';
-        print($outputFromIdp);
+        
+        $envelope = '<soap-env:Envelope xmlns:soap-env="http://schemas.xmlsoap.org/soap/envelope/">'.
+                    '<soap-env:Header/><soap-env:Body /></soap-env:Envelope>';
+
+        $doc = new DOMDocument();
+        $doc->loadXML($envelope);
+        $soapHeader = $doc->getElementsByTagNameNS('http://schemas.xmlsoap.org/soap/envelope/', 'Header');
+        $soapBody = $doc->getElementsByTagNameNS('http://schemas.xmlsoap.org/soap/envelope/', 'Body');
+
+        if ($message->toSignedXML()->localName === 'Response') {
```

Comment:
Did a bit of digging, and it looks the SOAP binding today is used for HTTP Artifact binding. `ArtifactResolutionService.php` creates a new `SOAP` binding, receives the artifact request, and responds with the stored artifact.

I believe the stored artifact could be a "Response" as an `AuthnRequest` can come in with the HTTP Artifact binding... I think this could break HTTP Artifact binding.